ipatests/test_integration/test_otp.py

```diff
@@ -309,3 +311,58 @@ def test_2fa_disable_single_prompt(self):
             master.run_command(['ipa', 'user-del', USER2])
             self.master.run_command(['semanage', 'login', '-D'])
             sssd_conf_backup.restore()
+
+    @pytest.fixture
+    def setup_otp_nsslapd(self):
+        # setting nsslapd-idletimeout
+        new_limit = 50
+        conn = self.master.ldap_connect()
+        dn = DN(('cn', 'config'))
+        entry = conn.get_entry(dn)  # pylint: disable=no-member
+        orig_limit = entry.single_value.get('nsslapd-idletimeout')
+        ldap_query = textwrap.dedent("""
+            dn: cn=config
+            changetype: modify
+            replace: nsslapd-idletimeout
+            nsslapd-idletimeout: {limit}
+        """)
+        tasks.ldapmodify_dm(self.master, ldap_query.format(limit=new_limit))
+        # Be sure no services are running
+        self.master.run_command(['killall', 'ipa-otpd'], raiseonerr=False)
+        yield
+        # cleanup
+        tasks.ldapmodify_dm(self.master, ldap_query.format(limit=orig_limit))
+
+    def test_check_otpd_after_dirsrv_restart(self, setup_otp_nsslapd):
+        """Test for ipa-otpd when dirsrv is restart with OTP auth.
+
+        Test for : https://pagure.io/freeipa/issue/6587
+
+        When OTP authentication is set and dirsrv restarted,
+        ipa-otpd service should not fail.
+        """
+        since = time.strftime('%H:%M:%S')
+        tasks.kinit_admin(self.master)
+        otpuid, totp = add_otptoken(self.master, USER, otptype="totp")
+        try:
+            # restart dirsrv.target
+            serverid = realm_to_serverid(self.master.domain.realm)
+            self.master.run_command(
+                ['systemctl', 'restart', 'dirsrv@%s.service' % serverid]
+            )
+
+            # kinit with OTP auth
+            otpvalue = totp.generate(int(time.time())).decode("ascii")
+            kinit_otp(self.master, USER, password=PASSWORD, otp=otpvalue)
+            time.sleep(60)
+            failed_services = self.master.run_command(
+                ['systemctl', 'list-units', '--state=failed']
+            )
```

Comment:
I wonder if you should run this first to be sure there are no failed units when the test starts.